src/AutoRest.CSharp/LowLevel/Decorator/MethodAmbiguityResolver.cs

```diff
@@ -0,0 +1,169 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using AutoRest.CSharp.Generation.Types;
+using AutoRest.CSharp.Input.Source;
+using AutoRest.CSharp.Output.Models;
+using AutoRest.CSharp.Output.Models.Shared;
+using AutoRest.CSharp.Output.Models.Types;
+using Microsoft.CodeAnalysis;
+
+namespace AutoRest.CSharp.LowLevel.Decorator
```

Comment:
Just curious, why it is a decorator?

Reply:
It's a common naming pattern when we want to modify the original model. It has something to do with the `decorator` pattern.

Reply:
I didn't see its relationship with `decorator` pattern. We don't have and additional behavior added to the initial object. Your implementation is more like a helper I think.

Reply:
If you check the other classes under `decorator` folders, you'll realize they don't strictly follow the classical `decorator pattern`. But the purpose of those classes is to decorate the existing models.

Here I follow the convention to put this class under `decorator` folder. Otherwise it could be put under the existing folder.

Reply:
Yeah, this is one of the things in mgmt that I don't want to bring into data plane. This isn't a decorator, and what is more important, this logic isn't optional. It has to be executed for every protocol signature that is going to be created, meaning that there is no reason to execute it after `LowLevelClientMethod` is instantiated. Instead, I'd create signature with correct set of parameters from the start, something linke:
```cs
var parameters = operation.IsLongRunning
    ? requestMethod.Parameters.Prepend(KnownParameters.WaitForCompletion).ToArray()
    : requestMethod.Parameters;

if (existingType is not null && existingType.GetMembers(requestMethod.Name).Any(s => s is IMethodSymbol method && IsAmbiguousButNotSame(method.Parameters, parameters)))
{
    parameters = parameters.Select(p => p with { DefaultValue = null }).ToArray();
}

var methodSignature = new MethodSignature(requestMethod.Name, requestMethod.Description, requestMethod.Accessibility | Virtual, returnType, null, parameters);
```